code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -825,4 +825,12 @@ var/list/ai_list = list()
 		user << "<span class='boldnotice'>Transfer successful</span>: [name] ([rand(1000,9999)].exe) removed from host terminal and stored within local memory."
 
 /mob/living/silicon/ai/flash_eyes(intensity = 1, override_blindness_check = 0, affect_silicon = 0)
-	return // no eyes, no flashing
\ No newline at end of file
+	return // no eyes, no flashing
+
+/mob/living/silicon/ai/attackby(obj/item/weapon/W, mob/user, params)
+	if(W.force && W.damtype != STAMINA) //only sparks if real damage is dealt.
+		var/datum/effect/effect/system/spark_spread/spark_system = new /datum/effect/effect/system/spark_spread

```

Comment:
make this a permanent var on the AI, like how some other machines and iirc borgs, do.
this way you just ```set_up()``` and ```start()``` every time instead of calling ```new```